Added default options method for RestController

GO3LIN opened the pull request:
Next to https://github.com/TurboGears/tg2/issues/90

amol- commented on tg/controllers/restcontroller.py:
```diff
@@ -68,5 +68,9 @@ class RestController(DecoratedController, CoreDispatcher, RestDispatcher):
     """
+    @expose()
+    def options(self):
+        return ""
```
Should return `None`, so that the response is a `204 No Content` instead of being a `200 OK` with no content.

amol- commented on tg/controllers/restcontroller.py:
```diff
@@ -68,5 +68,9 @@ class RestController(DecoratedController, CoreDispatcher, RestDispatcher):
 
     """
+    @expose()
+    def options(self):
```
Should probably accept **kwargs so that query parameters are ignored

amol- replied:
Also please add a docstring that states why this is here (like "here only to prevent 404 in CORS preflight requests, override it in your own controller to properly manage CORS")

amol- commented:
Please also add a test in https://github.com/TurboGears/tg2/blob/development/tests/test_rest_controller_dispatch.py

amol- commented:
And note that this is not working as `expose` is not available in that module -> https://travis-ci.org/TurboGears/tg2/jobs/260729192#L436

GO3LIN commented:
Thanks for your reading, I fixed everything you noticed in the comments and added a test case for the options method that just checks that the response status code is 204. Unfortunately, Travis failed on the python 3.6 build only.
